src/dim_helpers/DenseConvDims.jl

```diff
@@ -1,87 +1,103 @@
 export DenseConvDims
 
-"""
-    DenseConvDims
+struct DenseConvDims{N, K, S, P, D}
+    input_size::NTuple{N, Int}
 
-Concrete subclass of `ConvDims` for a normal, dense, conv2d/conv3d.
-"""
-struct DenseConvDims{N,K,C_in,C_out,G,S,P,D,F} <: ConvDims{N,S,P,D,F}
-    I::NTuple{N,Int}
-end
-
-# Getters for the fields
-input_size(c::DenseConvDims) = c.I
-kernel_size(c::DenseConvDims{N,K,C_in,C_out}) where {N,K,C_in,C_out} = K
-channels_in(c::DenseConvDims{N,K,C_in,C_out}) where {N,K,C_in,C_out} = C_in::Int
-channels_out(c::DenseConvDims{N,K,C_in,C_out,G}) where {N,K,C_in,C_out,G} = C_out::Int
-groupcount(c::DenseConvDims{N,K,C_in,C_out,G}) where {N,K,C_in,C_out,G} = G::Int
+    kernel_size::NTuple{K, Int}
+    channels_in::Int
+    channels_out::Int
+    groupcount::Int
 
-# Convenience wrapper to create DenseConvDims objects
-function DenseConvDims(x_size::NTuple{M}, w_size::NTuple{M};
-                       stride=1, padding=0, dilation=1, flipkernel::Bool=false, groups = 1) where M
+    stride::NTuple{S, Int}
+    padding::NTuple{P, Int}
+    dilation::NTuple{D, Int}
+    flipkernel::Bool
```

Comment:
```suggestion
    flipkernel::F
```
See comment thread above.